aws/resource_aws_key_pair.go

```diff
@@ -88,6 +94,30 @@ func resourceAwsKeyPairCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	d.SetId(*resp.KeyName)
+
+	if v := d.Get("tags").(map[string]interface{}); len(v) > 0 {
+		readReq := &ec2.DescribeKeyPairsInput{
+			KeyNames: []*string{aws.String(d.Id())},
+		}
+		readResp, err := conn.DescribeKeyPairs(readReq)
+		if err != nil {
+			awsErr, ok := err.(awserr.Error)
+			if ok && awsErr.Code() == "InvalidKeyPair.NotFound" {
+				d.SetId("")
+				return nil
+			}
+			return fmt.Errorf("Error retrieving KeyPair: %s", err)
+		}
+
+		for _, keyPair := range readResp.KeyPairs {
+			if *keyPair.KeyName == d.Id() {
+				if err := keyvaluetags.Ec2UpdateTags(conn, aws.StringValue(keyPair.KeyPairId), nil, v); err != nil {
```

Comment:
most of this annoying block is because we cant get the new `keyPairId` attribute from the `ImportKeyPair` Function